### DISCUSSION TITLE
Importance of this code.

### DISCUSSION THREAD

**Discussion by Knightrider16** [itsAnanth/xor-gate-perceptron#3] [Q&A] [answered]
Can you help in understanding the importance of this code

**Comment by itsAnanth** [comment, marked as answer]
This repo implements a simple MLP with 1 hidden layer from scratch, the primary goal is to just understand how neural networks work by looking at it from a much smaller scale. it includes how backpropogation works and how chain rule of calculus is applied to propogate gradients back to the gradients
